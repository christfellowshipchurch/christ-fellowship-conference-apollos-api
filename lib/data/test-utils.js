"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.buildGetMock = void 0;

/* eslint-disable import/prefer-default-export */
const buildGetMock = (response, dataSource) => {
  const get = jest.fn();

  if (Array.isArray(response) && Array.isArray(response[0])) {
    response.forEach(responseVal => {
      get.mockReturnValueOnce(new Promise(resolve => resolve(dataSource.normalize(responseVal))));
    });
  }

  get.mockReturnValue(new Promise(resolve => resolve(dataSource.normalize(response))));
  return get;
};

exports.buildGetMock = buildGetMock;
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIi4uLy4uL3NyYy9kYXRhL3Rlc3QtdXRpbHMuanMiXSwibmFtZXMiOlsiYnVpbGRHZXRNb2NrIiwicmVzcG9uc2UiLCJkYXRhU291cmNlIiwiZ2V0IiwiamVzdCIsImZuIiwiQXJyYXkiLCJpc0FycmF5IiwiZm9yRWFjaCIsInJlc3BvbnNlVmFsIiwibW9ja1JldHVyblZhbHVlT25jZSIsIlByb21pc2UiLCJyZXNvbHZlIiwibm9ybWFsaXplIiwibW9ja1JldHVyblZhbHVlIl0sIm1hcHBpbmdzIjoiOzs7Ozs7O0FBQUE7QUFDTyxNQUFNQSxZQUFZLEdBQUcsQ0FBQ0MsUUFBRCxFQUFXQyxVQUFYLEtBQTBCO0FBQ3BELFFBQU1DLEdBQUcsR0FBR0MsSUFBSSxDQUFDQyxFQUFMLEVBQVo7O0FBQ0EsTUFBSUMsS0FBSyxDQUFDQyxPQUFOLENBQWNOLFFBQWQsS0FBMkJLLEtBQUssQ0FBQ0MsT0FBTixDQUFjTixRQUFRLENBQUMsQ0FBRCxDQUF0QixDQUEvQixFQUEyRDtBQUN6REEsSUFBQUEsUUFBUSxDQUFDTyxPQUFULENBQWtCQyxXQUFELElBQWlCO0FBQ2hDTixNQUFBQSxHQUFHLENBQUNPLG1CQUFKLENBQ0UsSUFBSUMsT0FBSixDQUFhQyxPQUFELElBQWFBLE9BQU8sQ0FBQ1YsVUFBVSxDQUFDVyxTQUFYLENBQXFCSixXQUFyQixDQUFELENBQWhDLENBREY7QUFHRCxLQUpEO0FBS0Q7O0FBQ0ROLEVBQUFBLEdBQUcsQ0FBQ1csZUFBSixDQUNFLElBQUlILE9BQUosQ0FBYUMsT0FBRCxJQUFhQSxPQUFPLENBQUNWLFVBQVUsQ0FBQ1csU0FBWCxDQUFxQlosUUFBckIsQ0FBRCxDQUFoQyxDQURGO0FBR0EsU0FBT0UsR0FBUDtBQUNELENBYk0iLCJzb3VyY2VzQ29udGVudCI6WyIvKiBlc2xpbnQtZGlzYWJsZSBpbXBvcnQvcHJlZmVyLWRlZmF1bHQtZXhwb3J0ICovXG5leHBvcnQgY29uc3QgYnVpbGRHZXRNb2NrID0gKHJlc3BvbnNlLCBkYXRhU291cmNlKSA9PiB7XG4gIGNvbnN0IGdldCA9IGplc3QuZm4oKTtcbiAgaWYgKEFycmF5LmlzQXJyYXkocmVzcG9uc2UpICYmIEFycmF5LmlzQXJyYXkocmVzcG9uc2VbMF0pKSB7XG4gICAgcmVzcG9uc2UuZm9yRWFjaCgocmVzcG9uc2VWYWwpID0+IHtcbiAgICAgIGdldC5tb2NrUmV0dXJuVmFsdWVPbmNlKFxuICAgICAgICBuZXcgUHJvbWlzZSgocmVzb2x2ZSkgPT4gcmVzb2x2ZShkYXRhU291cmNlLm5vcm1hbGl6ZShyZXNwb25zZVZhbCkpKVxuICAgICAgKTtcbiAgICB9KTtcbiAgfVxuICBnZXQubW9ja1JldHVyblZhbHVlKFxuICAgIG5ldyBQcm9taXNlKChyZXNvbHZlKSA9PiByZXNvbHZlKGRhdGFTb3VyY2Uubm9ybWFsaXplKHJlc3BvbnNlKSkpXG4gICk7XG4gIHJldHVybiBnZXQ7XG59OyJdfQ==