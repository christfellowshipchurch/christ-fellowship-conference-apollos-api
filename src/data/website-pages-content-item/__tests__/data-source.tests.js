import ApollosConfig from '@apollosproject/config';
import WebsitePagesContentItem from '../data-source';
import { buildGetMock } from '../../test-utils'

ApollosConfig.loadJs({
    ROCK: {
        API_URL: 'https://apollosrock.newspring.cc/api',
        API_TOKEN: 'some-rock-token',
        IMAGE_URL: 'https://apollosrock.newspring.cc/GetImage.ashx',
    },
    ROCK_MAPPINGS: {
        WEBSITE_CONTENT_CHANNEL_IDS: {
            TEST_SITE: [1]
        }
    }
})

const website = 'TEST_SITE'
const title = 'hello-there'

describe('Website Pages Content Item', () => {
    // Test the normalizeSlug method
    it('returns a lowercase string with no spaces', () => {
        const dataSource = new WebsitePagesContentItem()
        const testCases = [
            { pass: 'Hello There', expect: 'hellothere' },
            { pass: 'Hello-There', expect: 'hello-there' }
        ]

        testCases.forEach(n => {
            const normalize = dataSource.normalizeSlug(n.pass)
            expect(normalize).toEqual(n.expect)
        })
    })

    // Test calls to the Content Channel Item Slugs table
    it('makes a call to ContentChannelItemSlugs where Slug is equal to the noramlized title passed in', async () => {
        const dataSource = new WebsitePagesContentItem()

        dataSource.get = jest.fn(() => Promise.resolve([{ ContentChannelItemId: 1, Slug: title }]))

        const result = await dataSource.getContentChannelIdByTitle(title)

        expect(dataSource.get.mock.calls).toMatchSnapshot()
    })

    it('makes a call to ContentChannelItemSlugs where Slug is not found and returns null', async () => {
        const dataSource = new WebsitePagesContentItem()

        dataSource.get = jest.fn(() => Promise.resolve([]))

        const result = await dataSource.getContentChannelIdByTitle(title)

        expect(result).toEqual(null)
    })

    // Test the possible return values when getting page content
    it('does not find slug for the title passed and returns null', async () => {
        const dataSource = new WebsitePagesContentItem()

        dataSource.getContentChannelIdByTitle = jest.fn(() => null)

        const result = await dataSource.getWebsitePageContentByTitle(website, title)

        expect(result).toEqual(null)
    })

    it('does not find a Content Channel Id for the website passed and returns null', async () => {
        const dataSource = new WebsitePagesContentItem()

        dataSource.get = jest.fn(() => Promise.resolve([{ ContentChannelItemId: 1, Slug: title }]))

        const result = await dataSource.getWebsitePageContentByTitle('NOT_THE_RIGHT_WEBSITE', title)

        expect(result).toEqual(null)
    })

    it('gets a Website Page Content Item from passing in a website key and page title', async () => {
        const dataSource = new WebsitePagesContentItem()

        dataSource.get = buildGetMock([
            [{ ContentChannelItemId: 1, Slug: title }],
            [{ Id: 1, ContentChannelId: 10 }]
        ], dataSource)

        const result = await dataSource.getWebsitePageContentByTitle(website, title)

        expect(dataSource.get.mock.calls).toMatchSnapshot();
    })
});