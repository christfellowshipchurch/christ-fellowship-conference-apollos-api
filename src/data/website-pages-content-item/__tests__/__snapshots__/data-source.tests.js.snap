// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Website Pages Content Item gets a Website Page Content Item from passing in a website key and page title 1`] = `
Array [
  Array [
    "ContentChannelItemSlugs?loadAttributes=expanded&%24filter=Slug%20eq%20%27hello-there%27&%24select=ContentChannelItemId%2C%20Slug&%24top=1",
    Object {},
    Object {},
  ],
  Array [
    "ContentChannelItems?loadAttributes=expanded&%24filter=%28%28ContentChannelId%20eq%201%29%29%20and%20%28Id%20eq%201%29&%24top=1",
    Object {},
    Object {},
  ],
]
`;

exports[`Website Pages Content Item makes a call to ContentChannelItemSlugs where Slug is equal to the noramlized title passed in 1`] = `
Array [
  Array [
    "ContentChannelItemSlugs?loadAttributes=expanded&%24filter=Slug%20eq%20%27hello-there%27&%24select=ContentChannelItemId%2C%20Slug&%24top=1",
    Object {},
    Object {},
  ],
]
`;
